definitions/npm/async_v1.x.x/flow_>=v0.28.x/async_v1.x.x.js

```diff
@@ -0,0 +1,77 @@
+// @flow
+
+declare module 'async' {
+  declare type Collection<TItem> = Iterable<TItem> | { [key: string]: TItem };
+
+  declare type IterateeCallback<TError, TResult> = (err?: TError, result?: TResult) => void;
+  declare type Callback<TError, TResult> = (err?: TError, result?: TResult) => mixed;
+
+  // Collections
+  declare function each<TItem, TError>(
+    collection: Collection<TItem>,
+    iteratee: (item: TItem, callback: IterateeCallback<TError, void>) => mixed,
+    callback?: Callback<TError, void>
+  ): void;
+  declare function forEachOf<TItem, TError>(
+    collection: Collection<TItem>,
+    iteratee: (value: TItem, key: string | number, callback: IterateeCallback<TError, void>) => mixed,
```

Comment:
Here you can see for `key` I am using `string | number`, but I would prefer to use a polymorphic type that pulls from the abstract `Collection` type (which accepts anything that implements an iterator, as well as objects).

Here is a table of the mapping of different values of `collection` to `value` and `key`:


| collection | value | key |
|:--- |:--- |:--- |
| `Array [a]` | `a` | `0` |
| `Object { a: b }` | `b` | `a` |
| `Map { a => b }` | `[a, b]` | `0` |
| `Set { a }` | `a` | `0` |

Note that `Iterable<[TKey, TValue]>` is not what I am looking for... that doesn't work for sets and doesn't match the way maps work as noted in the table above.